### ISSUE TITLE
Signature benchmark

### ISSUE THREAD

**Issue by stemangiola** [stemangiola/cellsig#38]

**Comment by stemangiola** [comment]
1) review Jian's code, and identify testing-with-deconvolution routine
2) use it to obtain error scores

here you can start apply this to your KN project

once you completed that ..

3) design a more proper test
- GENERALISE TO NEW DATASETS build a function that creates a training set, and a test set (includes one dataset for each cell type), build an in silico mixture (Jian should have all the code) with the test dataset, build the signature with the remaining training data, deconvolvolve and evaluate error on proportions, repeat 3 times to start
- TESTING MARKER FAILURE build a benchmark where we perturb 10% of genes of the signature (in this case you could test on training set)

**Comment by stemangiola** [comment]
For database cross-validation, just one subsampling, 2 mixtures, and three algorithms = 6 

This is the brought pipeline

<img width="1084" alt="image" src="https://user-images.githubusercontent.com/7232890/167791766-5a123085-9b8f-4acd-bafb-d78f83507046.png">

**Comment by stemangiola** [comment]
Hello @Kamran-Khan96 , for the gene perturbation benchmark, I found a quite good representation of results in this paper. The size of the circles represents the error derived by removing (or perturbing) a gene.

https://www.nature.com/articles/s41467-022-28020-5

<img width="716" alt="image" src="https://user-images.githubusercontent.com/7232890/169948063-e2ff97b9-a113-4415-94b2-e074b646d223.png">

**Comment by Kamran-Khan96** [comment]
> Hello @Kamran-Khan96 , for the gene perturbation benchmark, I found a quite good representation of results in this paper. The size of the circles represents the error derived by removing (or perturbing) a gene.
> 
> https://www.nature.com/articles/s41467-022-28020-5
> 
> <img alt="image" width="716" src="https://user-images.githubusercontent.com/7232890/169948063-e2ff97b9-a113-4415-94b2-e074b646d223.png">

Ah, looks interesting! Specially, we can easily create figures like the figure 1a. However, for 1b, I think they used two observed proportion values in the 2 axes, whereas we have 23 different variables. Maybe we can do pairwise plots for the different cell types, or focus on the more critical pairs, for instance- CD8-T cells and NK-cells. But this is a very great approach of detailing the information. Thanks, we can obviously try to incorporate figures like these.

**Comment by Kamran-Khan96** [comment]
Obsolete
